libwild/src/output_section_id.rs

```diff
@@ -636,6 +637,7 @@ const SECTION_DEFINITIONS: [BuiltInSectionDetails; NUM_BUILT_IN_SECTIONS] = [
         kind: SectionKind::Primary(SectionName(TEXT_SECTION_NAME)),
         ty: sht::PROGBITS,
         section_flags: shf::ALLOC.with(shf::EXECINSTR),
+        start_symbol_name: Some(EXECUTABLE_START),
         ..DEFAULT_DEFS
```

Comment:
This is wrong because that symbols should be emitted only for (some kinds of?) executables. Probably should be created inside `Prelude::new`.

Similarly, Wild outputs `__ehdr_start` even when other linkers don't, and it also misses `__dso_handle` symbol.

Reply:
I haven't looked into this too much, but a couple of things that you were probably aware of, but just in case... some of these symbols are `PROVIDE` in the GNU ld linker script, which AFAIK means that the linker script should only define the symbol if it's referenced and not defined by anything input objects. We don't yet support PROVIDE semantics, although it might be being worked on - #1097.

Some flags can affect the built-in linker script that GNU ld uses. You can see what changes for some particular flag with a command like `meld <(ld --verbose) <(ld --pie --verbose)`.

Reply:
Thanks, I didn't see your comment when I was writing the other one (GitHub didn't refresh automatically). Well, I was unaware but figured that out eventually.